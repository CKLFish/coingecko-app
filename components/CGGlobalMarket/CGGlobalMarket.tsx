import React from 'react';
import {useTranslation} from 'react-i18next';
import {Dimensions, StyleSheet, View} from 'react-native';
import {PieChart} from 'react-native-chart-kit';
import {AbstractChartConfig} from 'react-native-chart-kit/dist/AbstractChart';
import {PieChartProps} from 'react-native-chart-kit/dist/PieChart';
import {Card, Text} from 'react-native-paper';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import {TextUtil} from '../../Utils';
import CGColorText from '../CGColorText';

export type MarketCapPercentageList = {
  value: number;
  name: string;
  color: string;
  legendFontColor: string;
  legendFontSize: number;
};

export type CGGlobalMarketProps = {
  onGoingIcos: number;
  endedIcos: number;
  upcomingIcos: number;
  activeCryptoCurrencies: number;
  iconSize: number;
  currencySign?: string;
  totalMarketCap: number;
  marketCapPercentageList: MarketCapPercentageList[];
  marketCapChangedPercentage: number;
  marketCapPercantageChartProps?: Partial<PieChartProps>;
};

const defaultChartConfig: AbstractChartConfig = {
  backgroundGradientFrom: '#1E2923',
  backgroundGradientFromOpacity: 0,
  backgroundGradientTo: '#08130D',
  backgroundGradientToOpacity: 0.5,
  color: (opacity = 1) => `rgba(26, 255, 146, ${opacity})`,
  strokeWidth: 2,
  barPercentage: 0.5,
};

const CGGlobalMarket: React.FC<CGGlobalMarketProps> = props => {
  const {
    onGoingIcos,
    endedIcos,
    upcomingIcos,
    activeCryptoCurrencies,
    totalMarketCap,
    marketCapPercentageList,
    marketCapChangedPercentage,
    marketCapPercantageChartProps,
    currencySign = '',
    iconSize = 30,
  } = props;
  const {chartConfig, ...otherChartProps} = marketCapPercantageChartProps || {};
  const {t} = useTranslation();

  return (
    <View>
      <View style={[styles.row, styles.flex]}>
        <Card style={[styles.flex, styles.sharpCorner]}>
          <Card.Title
            title={`${onGoingIcos}`}
            subtitle={t('CRYPTO_MARKET_ICO_ONGOING')}
            left={() => (
              <MaterialCommunityIcons name="file-clock" size={iconSize} />
            )}
          />
        </Card>
        <Card style={[styles.flex, styles.sharpCorner]}>
          <Card.Title
            title={`${endedIcos}`}
            subtitle={t('CRYPTO_MARKET_ICO_ENDED')}
            left={() => (
              <MaterialCommunityIcons name="file-check" size={iconSize} />
            )}
          />
        </Card>
      </View>
      <View style={[styles.row, styles.flex]}>
        <Card style={[styles.flex, styles.sharpCorner]}>
          <Card.Title
            title={`${upcomingIcos}`}
            subtitle={t('CRYPTO_MARKET_ICO_UPCOMING')}
            left={() => (
              <MaterialCommunityIcons name="file-plus" size={iconSize} />
            )}
          />
        </Card>
        <Card style={[styles.flex, styles.sharpCorner]}>
          <Card.Title
            title={activeCryptoCurrencies}
            subtitle={t('CRYPTO_MARKET_ACTIVE_CRYPTO')}
            left={() => (
              <MaterialCommunityIcons name="bitcoin" size={iconSize} />
            )}
          />
        </Card>
      </View>
      <Card style={styles.sharpCorner}>
        <Card.Title
          title={`${currencySign}${
            totalMarketCap
              ? TextUtil.formatCurrency(totalMarketCap)
              : t('EMPTY_PLACEHOLDER')
          }`}
          subtitle={`${t('CRYPTO_MARKET_TOTAL_MARKET_CAP')}`}
          left={() => <MaterialCommunityIcons name="cash" size={iconSize} />}
          right={() =>
            marketCapChangedPercentage ? (
              <CGColorText value={marketCapChangedPercentage} />
            ) : (
              <Text>{t('EMPTY_PLACEHOLDER')}</Text>
            )
          }
        />
      </Card>
      <Card style={styles.sharpCorner}>
        <Card.Title title={t('CRYPTO_MARKET_PIECHART_TITLE')} />
        {marketCapPercentageList ? (
          <PieChart
            width={Dimensions.get('window').width}
            height={220}
            backgroundColor={'transparent'}
            paddingLeft={'14'}
            chartConfig={{
              ...defaultChartConfig,
              ...chartConfig,
            }}
            {...otherChartProps}
            data={marketCapPercentageList}
            accessor={'value'}
          />
        ) : null}
      </Card>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  flex: {
    flex: 1,
  },
  sharpCorner: {
    borderRadius: 0,
  },
  row: {
    flexDirection: 'row',
  },
});

export default CGGlobalMarket;
