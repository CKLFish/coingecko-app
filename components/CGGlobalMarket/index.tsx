import CGGlobalMarket from './CGGlobalMarket';

export default CGGlobalMarket;
