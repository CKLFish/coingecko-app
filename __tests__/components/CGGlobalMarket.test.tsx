/**
 * @format
 */

import 'react-native';

import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import CGGlobalMarket from '../../components/CGGlobalMarket';

describe('CGGlobalMarket', () => {
  test('0.0', () => {
    const globalMarketInstance = renderer.create(
      <CGGlobalMarket
        onGoingIcos={10}
        endedIcos={10}
        upcomingIcos={10}
        activeCryptoCurrencies={144}
        totalMarketCap={40}
        marketCapChangedPercentage={50}
        marketCapPercentageList={[
          {
            name: 'a',
            value: 50,
            color: '#666666',
            legendFontColor: '#666666',
            legendFontSize: 14,
          },
          {
            name: 'b',
            value: 50,
            color: '#99ffff',
            legendFontColor: '#99ffff',
            legendFontSize: 14,
          },
        ]}
        currencySign={'$'}
        iconSize={30}
      />,
    );
    expect(globalMarketInstance.toJSON()).toMatchSnapshot();
  });
});
