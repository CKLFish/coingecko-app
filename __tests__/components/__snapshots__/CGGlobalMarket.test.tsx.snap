// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CGGlobalMarket 0.0 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.12)",
          "borderRadius": 0,
          "elevation": 1,
          "flex": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "minHeight": 72,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󱋡
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              10
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                          "paddingRight": 16,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              CRYPTO_MARKET_ICO_ONGOING
            </Text>
          </View>
          <View />
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.12)",
          "borderRadius": 0,
          "elevation": 1,
          "flex": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "minHeight": 72,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰈖
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              10
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                          "paddingRight": 16,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              CRYPTO_MARKET_ICO_ENDED
            </Text>
          </View>
          <View />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.12)",
          "borderRadius": 0,
          "elevation": 1,
          "flex": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "minHeight": 72,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰝒
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              10
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                          "paddingRight": 16,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              CRYPTO_MARKET_ICO_UPCOMING
            </Text>
          </View>
          <View />
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.12)",
          "borderRadius": 0,
          "elevation": 1,
          "flex": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "minHeight": 72,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰠓
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              144
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                          "paddingRight": 16,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              CRYPTO_MARKET_ACTIVE_CRYPTO
            </Text>
          </View>
          <View />
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 0,
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            Object {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 40,
                "justifyContent": "center",
                "marginRight": 16,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰄔
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "minHeight": 30,
                        "paddingRight": 16,
                      },
                      Object {
                        "marginBottom": 0,
                      },
                      undefined,
                    ],
                  ],
                ],
              ]
            }
          >
            $40.00
          </Text>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "marginVertical": 0,
                        "minHeight": 20,
                        "paddingRight": 16,
                      },
                      undefined,
                    ],
                  ],
                ],
              ]
            }
          >
            CRYPTO_MARKET_TOTAL_MARKET_CAP
          </Text>
        </View>
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": undefined,
                },
                undefined,
              ]
            }
          >
            50.0%
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 0,
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            Object {
              "minHeight": 50,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {
                        "minHeight": 30,
                        "paddingRight": 16,
                      },
                      Object {
                        "marginBottom": 2,
                      },
                      undefined,
                    ],
                  ],
                ],
              ]
            }
          >
            CRYPTO_MARKET_PIECHART_TITLE
          </Text>
        </View>
        <View />
      </View>
      <View
        style={
          Object {
            "height": 220,
            "padding": 0,
            "width": 750,
          }
        }
      >
        <RNSVGSvgView
          bbHeight={220}
          bbWidth={750}
          focusable={false}
          height={220}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 220,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      1976611,
                      1,
                      -2146954483,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="backgroundGradient"
                  x1={0}
                  x2={220}
                  y1={220}
                  y2={0}
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      437976978,
                      1,
                      1769362,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="fillShadowGradient"
                  x1={0}
                  x2={0}
                  y1={0}
                  y2={220}
                />
              </RNSVGDefs>
            </RNSVGGroup>
            <RNSVGRect
              fill={0}
              height={220}
              propList={
                Array [
                  "fill",
                ]
              }
              rx={0}
              ry={0}
              width="100%"
              x={0}
              y={0}
            />
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  201.5,
                  110,
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M 0 -88 A 88 88 0 0 1 0 88 L 0 0 A 0 0 0 0 0 0 0 Z "
                  fill={4284900966}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGRect
                  fill={4284900966}
                  height="16px"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rx={8}
                  ry={8}
                  width="16px"
                  x={276}
                  y={-76}
                />
                <RNSVGText
                  content={null}
                  dx={Array []}
                  dy={Array []}
                  fill={4284900966}
                  font={
                    Object {
                      "fontSize": 14,
                    }
                  }
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotate={Array []}
                  x={
                    Array [
                      300,
                    ]
                  }
                  y={
                    Array [
                      -64,
                    ]
                  }
                >
                  <RNSVGTSpan
                    content="50% a"
                    dx={Array []}
                    dy={Array []}
                    font={Object {}}
                    rotate={Array []}
                    x={Array []}
                    y={Array []}
                  />
                </RNSVGText>
              </RNSVGGroup>
              <RNSVGGroup>
                <RNSVGPath
                  d="M 0 88 A 88 88 0 0 1 -0.0088 -88 L 0 0 A 0 0 0 0 0 0 0 Z "
                  fill={4288282623}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGRect
                  fill={4288282623}
                  height="16px"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rx={8}
                  ry={8}
                  width="16px"
                  x={276}
                  y={12}
                />
                <RNSVGText
                  content={null}
                  dx={Array []}
                  dy={Array []}
                  fill={4288282623}
                  font={
                    Object {
                      "fontSize": 14,
                    }
                  }
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotate={Array []}
                  x={
                    Array [
                      300,
                    ]
                  }
                  y={
                    Array [
                      24,
                    ]
                  }
                >
                  <RNSVGTSpan
                    content="50% b"
                    dx={Array []}
                    dy={Array []}
                    font={Object {}}
                    rotate={Array []}
                    x={Array []}
                    y={Array []}
                  />
                </RNSVGText>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
